### ISSUE TITLE
Report superfluous "return" uses

### ISSUE THREAD

**Issue by vpelletier** [koalaman/shellcheck#1995]
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
foo () {
  true
  return $?
}

```

#### Here's what shellcheck currently says:

```
No issues detected!
```

#### Here's what I wanted or expected to see:

Possibly with some optional checker(s) enabled:
- a warning about `$?` being superfluous, as it is the implicit default (or is it just in bash ? I could not locate posix shell specification on return statement)
- a warning about `return` being superfluous, as it is the last statement before function end (ideally, it would also be reported in any nested bloc when there is no other statement until function end)

I think the former is probably good to enable by default. The latter is probably best left to be explicitly enabled.

**Comment by e-kwsm** [comment]
related (`$?`): #2886
